File: app/src/main/java/com/attendo/data/sub/SubEntity.java
package com.attendo.data.sub;


import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "SubjectName")
public class SubEntity
{
    @PrimaryKey(autoGenerate = true)
    @NonNull
    private int id;

    @NonNull
    @ColumnInfo(name = "subject")
    private String subject;

    @ColumnInfo(name = "present")
    private int present;

    @ColumnInfo(name="absent")
    private int absent;

    @ColumnInfo(name = "total")
    private int total;

    public SubEntity( @NonNull String subject, int present,int absent, int total) {
        this.subject= subject;
        this.present = present;
        this.total = total;
        this.absent = absent;
    }

    @NonNull
    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPresent() {
        return present;
    }

    public void setPresent(int present) {
        this.present = present;
    }

    public int getAbsent() {
        return absent;
    }

    public void setAbsent(int absent) {
        this.absent = absent;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }
}

File: app/src/main/java/com/attendo/data/sub/SubEntity.kt
package com.attendo.data.sub

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "SubjectName")
public class SubEntity(
    @ColumnInfo(name = "subject")
    private var subject: String?, @ColumnInfo(name = "present")
    private var present: Int, @ColumnInfo(name = "absent")
    private var absent: Int, @ColumnInfo(name = "total")
    private var total: Int
) {

    @PrimaryKey(autoGenerate = true)
    private var id: Int? = 0

    fun getSubject(): String? {
        return subject
    }

    fun setSubject(subject: String) {
        this.subject = subject
    }

    fun getId(): Int? {
        return id
    }

    fun setId(id: Int) {
        this.id = id
    }

    fun getPresent(): Int {
        return present
    }

    fun setPresent(present: Int) {
        this.present = present
    }

    fun getAbsent(): Int {
        return absent
    }

    fun setAbsent(absent: Int) {
        this.absent = absent
    }

    fun getTotal(): Int {
        return total
    }

    fun setTotal(total: Int) {
        this.total = total
    }
}